dhcp6/packet_builder.go

```diff
@@ -0,0 +1,174 @@
+package dhcp6
+
+import (
+	"hash/fnv"
+	"encoding/binary"
+	"net"
+)
+
+// PacketBuilder is used for generating responses to requests received from dhcp clients
+type PacketBuilder struct {
+	PreferredLifetime uint32
+	ValidLifetime     uint32
+}
+
+// MakePacketBuilder creates a new PacketBuilder and initializes it with preferred and valid lifetimes
+func MakePacketBuilder(preferredLifetime, validLifetime uint32) *PacketBuilder {
+	return &PacketBuilder{PreferredLifetime: preferredLifetime, ValidLifetime: validLifetime}
+}
+
+// BuildResponse generates a response packet for a packet received from a client
+func (b *PacketBuilder) BuildResponse(in *Packet, serverDUID []byte, configuration BootConfiguration, addresses AddressPool) (*Packet, error) {
```

Comment:
> should this logic live in the pixiecore package instead? It looks like it's dedicated to constructing a reply for netbooting, so it shouldn't be in the generic DHCPv6 package.

Hmm. FWIW, this isn't meant to be used as a generic dhcpv6 server, it will ignore dhcp requests without boot file url option request present. I don't think we could remove any logic that is currently there either -- the implementation is pretty minimal.

To me netbooting is part of dhcpv6, not sure it's worth it to try to extract the logic that is used to populate netbooting-specific options into a different package. I moved ```BootConfiguration``` implementations into the ```pixicorev6``` package already -- I think this separates pixicore from dhcp quite well?